### PR TITLE
Add checks for publishing Focus apks

### PR DESCRIPTION
**Pull request by liuche**
Adds checks for:
* package names matching Focus/Klar
* 2 apks max
* distinct package names

### CODE REVIEW

**Comment by liuche** [comment]
```
2018-04-26 10:59:56,348 - extractor.py - INFO - Extracting metadata from a copy of "/Users/liuche/Desktop/Focus.apk"...
2018-04-26 10:59:56,540 - extractor.py - INFO - Extracting metadata from a copy of "/Users/liuche/Desktop/Klar.apk"...
2018-04-26 10:59:56,713 - checker.py - INFO - Checking APKs' metadata and content...
2018-04-26 10:59:56,714 - checker.py - INFO - All APKs have distinct Package name: ['org.mozilla.klar', 'org.mozilla.focus']
2018-04-26 10:59:56,715 - checker.py - INFO - Found expected number of APKs, not more than 2
2018-04-26 10:59:56,717 - checker.py - INFO - Expected product types [<PRODUCT.FOCUS: 'org.mozilla.focus'>, <PRODUCT.KLAR: 'org.mozilla.klar'>] found
2018-04-26 10:59:56,718 - checker.py - INFO - APKs are sane!
2018-04-26 10:59:56,719 - push_apk.py - WARNING - Listing and what's new section won't be updated.
2018-04-26 10:59:56,723 - googleplay.py - WARNING - `--do-not-contact-google-play` option was given. Not a single request to Google Play will be made!
2018-04-26 10:59:56,724 - googleplay.py - INFO - Uploading "/Users/liuche/Desktop/Focus.apk"
2018-04-26 10:59:56,725 - googleplay.py - INFO - "/Users/liuche/Desktop/Focus.apk" uploaded
2018-04-26 10:59:56,726 - googleplay.py - DEBUG - Upload response: {'versionCode': 'fake-version-code'}
2018-04-26 10:59:56,727 - googleplay.py - INFO - Uploading "/Users/liuche/Desktop/Klar.apk"
2018-04-26 10:59:56,727 - googleplay.py - INFO - "/Users/liuche/Desktop/Klar.apk" uploaded
2018-04-26 10:59:56,728 - googleplay.py - DEBUG - Upload response: {'versionCode': 'fake-version-code'}
2018-04-26 10:59:56,729 - googleplay.py - INFO - Track "internal" updated with: {'versionCodes': ['20811502', '20811502']}
2018-04-26 10:59:56,730 - googleplay.py - DEBUG - Track update response: fake-update-response
2018-04-26 10:59:56,731 - googleplay.py - WARNING - `commit` option was not given. Transaction not committed.
```

**Comment by liuche** [comment]
Ah, I will fix the lint errors!

File: mozapkpublisher/common/apk/checker.py [hunk 1]
```diff
@@ -67,6 +74,38 @@ def cross_check_apks(apks_metadata_per_paths):
...
+        raise BadSetOfApks('Expected product types {}, found {}'.format(product_types, types))
+    logger.info('Expected product types {} found'.format(product_types))
+
+
+def _check_piece_of_metadata_is_distinct(key, pretty_key, apks_metadata_per_paths):
+    all_items = [metadata[key] for metadata in apks_metadata_per_paths.values()]
+    unique_items = filter_out_identical_values(all_items)
+
+    if not unique_items:
+        raise BadSetOfApks('No {} found')
```
Comment (JohanLorenzo) [inline mozapkpublisher/common/apk/checker.py]: Nit: missing `.format()`

File: mozapkpublisher/test/test_push_apk.py [hunk 2]
```diff
@@ -220,6 +221,25 @@ def test_upload_apk_with_locales_updated_from_file(edit_service_mock, monkeypatc
...
 
 
+def test_get_distinct_package_name_apk_metadata():
+    one_package_apks_metadata = {
+        'fennec-1.apk': {'package_name': 'org.mozilla.firefox'},
+        'fennec-2.apk': {'package_name': 'org.mozilla.firefox'}
+    }
+
+    one_metadata = _get_distinct_package_name_apk_metadata(one_package_apks_metadata)
+    assert len(one_metadata.keys()) == 1
```
Comment (JohanLorenzo) [inline mozapkpublisher/test/test_push_apk.py]: Discussed offline. I got confused by the expected output of this function. In order to make clearer, we could add an extra assertion to show we still have the 2 fennecs apks in `one_metadata[package_name]`.

File: mozapkpublisher/push_apk.py [hunk 3]
```diff
@@ -77,22 +77,35 @@ def run(self):
...
-            l10n_strings = store_l10n.get_translations_per_google_play_locale_code(package_name)
-        elif not self.config.update_google_play_strings:
-            logger.warn("Listing and what's new section won't be updated.")
-            l10n_strings = None
-        else:
-            raise WrongArgumentGiven("Option missing. You must provide what to do in regards to Google Play strings.")
-
-        self.upload_apks(apks_metadata_per_paths, package_name, l10n_strings)
+
+        package_apk_metadata = _get_distinct_package_name_apk_metadata(apks_metadata_per_paths)
```
Comment (JohanLorenzo) [inline mozapkpublisher/push_apk.py]: After giving it some thought, we could name this function `_split_apk_metadata_per_package_name()` and then leave a comment like 
``` py
# We must upload different products in different Google Play transaction. That's why we're splitting them at this level"
```

File: mozapkpublisher/push_apk.py [hunk 4]
```diff
@@ -77,22 +77,35 @@ def run(self):
...
+
+        for (package_name, apks_metadata) in package_apk_metadata.items():
+            if self.config.google_play_strings_file:
+                l10n_strings = json.load(self.config.google_play_strings_file)
+                store_l10n.check_translations_schema(l10n_strings)
+                logger.info('Loaded listings and what\'s new section from "{}"'.format(self.config.google_play_strings_file.name))
+            elif self.config.update_google_play_strings_from_store:
+                logger.info("Downloading listings and what's new section from L10n Store...")
+                l10n_strings = store_l10n.get_translations_per_google_play_locale_code(package_name)
```
Comment (JohanLorenzo) [inline mozapkpublisher/push_apk.py]: Good call on fetching different strings dependending on the package name. I see Klar is defined as a different production in https://l10n.mozilla-community.org/stores_l10n/locale/de/klar_android/release/ as well.

Firefox focus equivalent: https://l10n.mozilla-community.org/stores_l10n/locale/de/focus_android/release/

File: mozapkpublisher/test/common/apk/test_checker.py [hunk 5]
```diff
@@ -50,6 +53,52 @@ def test_cross_check_apks():
...
+        'Klar.apk': {
+            'api_level': 21,
+            'architecture': 'armeabi-v7',
+            'package_name': 'org.mozilla.klar',
+            'version_code': '11'
+        }
+    })
+
+
+def test_check_number_of_apks():
```
Comment (JohanLorenzo) [inline mozapkpublisher/test/common/apk/test_checker.py]: Nice tests in this file :+1:

**Comment by liuche** [comment]
Changed check for number of apks to check number of packages in order to handle multiple architectures for a single product/package.

**Pull request merged**